codebase/src/graph/graphAnimation.ts

```diff
@@ -23,13 +45,35 @@ const animateGraph = (nodes: GNode[], edges: GEdge[], trace: number[][]) => {
     );
 
     if (edge) {
-      edge.status = Status.exploring;
-      nextNode!.status = Status.exploring;
+      edge.status = "exploring";
+      nextNode!.status = "exploring";
```

Comment:
we shouldnt be using non-null assertions here. Handle possible undefined next node by narrowing the type of nextNode with a `guard clause`